qcodes/instrument_drivers/QuTech/S5i.py

```diff
@@ -20,34 +20,88 @@ class S5i(Instrument):
             individual modules.
 
         module (int): module number as set on the hardware.
+
+    Parameters:
+        enabled: switched output on/off
+        frequency: Frequency on Hz, range 40 MHz - 4 GHz, stepsize 1 MHz
+        freq_fine: Frequency on Hz, range 40 MHz - 4 GHz, stepsize 10 kHz.
+        Difference between frequency and freq_fine unclear!?
+        power: power in dBm, range -20 to +15 dBm
+
     """
 
     def __init__(self, name, spi_rack, module, **kwargs):
         super().__init__(name, **kwargs)
 
-        self.s5i = S5i_module(spi_rack, module)
-
-        self.add_parameter('use_external_reference',
-                           label='Use external reference',
-                           get_cmd=self._use_external_reference,
-                           set_cmd=self.s5i.use_external_reference,
+        self.s5i = S5i_module(spi_rack, module, frequency=41e6,
```

Comment:
is this change being introduced because a new version of `spirack` has been released? if so, shall we make it clear somewhere (perhaps in requirements.txt or environment.yml, or just in the docstring of this driver) that a particular version of `spirack` is required?

Reply:
No, the spirack driver is the same. But we choose to have different default settings. The `spirack` is not yet in the `requirements.txt`, since the spirack is optional. Is there another way to specify version requirements?

Reply:
well, i do not think we have a good approach to this. Ideally every python module has its setup.py file where the required packages and their versions are listed. But at the moment all drivers are part of qcodes, hence they don't have their independent setup.py files.

perhaps, in the exception that is thrown in case `spirack` is not installed, you can mention the version requirements like "install spirack >=1.2.3".